diff -up asio-1.4.1/include/asio/ssl/detail/openssl_context_service.hpp\~ asio-1.4.1/include/asio/ssl/detail/openssl_context_service.hpp
--- asio-1.4.1/include/asio/ssl/detail/openssl_context_service.hpp~	2009-03-03 03:22:54.000000000 +0100
+++ asio-1.4.1/include/asio/ssl/detail/openssl_context_service.hpp	2010-02-23 15:54:15.000000000 +0100
@@ -66,7 +66,7 @@ public:
   // Create a new context implementation.
   void create(impl_type& impl, context_base::method m)
   {
-    ::SSL_METHOD* ssl_method = 0;
+    const ::SSL_METHOD* ssl_method = 0;
     switch (m)
     {
     case context_base::sslv2:
